### PR TITLE
Allow users to select a course regardless of if some sections are full or open

### LINKED ISSUE #377: Clicking a listing's title should select it
~Currently, clicking a listing's title does not select it. This is confusing and should be fixed ASAP. Additionally, the course should not highlight *unless* the user is hovering over an area that will select the listing if clicked.~

Current behavior: When a user clicks a course to select it, the course is not selected if all of the sections are full, because we do not by default select full sections. This is by design, the user is intended to click full sections individually to select them.

HOWEVER, we have received much feedback that this is confusing, and many users think selection is broken when they cannot click a course due to all its sections being full. To relieve this, I think we should do away with the notion of not selecting full sections.

Proposed new behavior: When a user clicks a course, all sections should be selected regardless of if they are full or open. This should avoid confusion, and prevent our users from thinking they cannot select closed sections, which they can.

Additionally, it would be nice if we showed if a section was closed in the schedule view somehow, maybe by graying out the block. But we can create a separate issue for that.

### PR DESCRIPTION
**Pull request by fakedestinyck**
Resolves #377.
Some previews:
image
image

### CODE REVIEW

File: web/src/app/services/selection.service.ts [hunk 1]
```diff
@@ -55,19 +55,22 @@ export class SelectionService {
...
     if (this.hasSelectedSection(course)) {
       let store = this.getSelections();
       delete store[course.id];
       this.setItem('selections', JSON.stringify(store));
     } else {
+      closedCount = 0;
       course.sections.forEach((s) => {
-        if (s.seatsTaken < s.seats) {
-          this.addSection(s);
+        if (s.seatsTaken >= s.seats) {
```
Comment (Bad-Science) [inline web/src/app/services/selection.service.ts]: I don't think this logic should go here. The number of closed sections isn't really related to toggling a course, so I think that should go somewhere else, probably the course component

File: web/src/app/listing/component.html [hunk 2]
```diff
@@ -4,9 +4,13 @@
 	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered && !isCourseSelected()" [class.indicator-selected-hover]="hovered && isCourseSelected()"></div>
 	  </div>
 	  <div class="w-100 h-100 course-content position-relative min-height">
-		  <table (click)="clickCourse()" class="w-100">
+		  <!-- <img *ngIf="getSectionClosedStatus() === 'someClosed'" src="assets/images/some-closed.png" class="warning" placement="right" ngbTooltip="Some sections are closed">
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: Remove commented code

Comment (fakedestinyck) [reply web/src/app/listing/component.html]: Removed @Bad-Science

File: web/src/app/listing/component.ts [hunk 3]
```diff
@@ -75,7 +75,41 @@ export class ListingComponent implements OnInit{
   }
 
   public get tooltipDescription (): string {
-    return this.listing.description || 'No description available :('; 
+    return this.listing.description || 'No description available :(';
+  }
+
+  public getSectionClosedStatus (): string {
```
Comment (Bad-Science) [inline web/src/app/listing/component.ts]: Avoid using "magic strings" as return values from functions. We can refactor this to make it cleaner. I suggest: moving the counting logic to its own function `getClosedSectionCount`, and comparing the closed count to the number of listings in a separate function, or even in the view. Then you won't need the `"allClosed"` and `"someClosed"` strings. The same goes for conflicts.

Comment (fakedestinyck) [reply web/src/app/listing/component.ts]: Fixed! @Bad-Science

**Comment by Bad-Science** [comment]
Look good! Thanks for all the work @fakedestinyck !!

**Pull request merged**